### PR TITLE
Make the position parameter work properly

### PR DESCRIPTION
**Pull request by mmarod**
This fixes issue #3 by adding self.position_path and in_position? methods to the provider.

### CODE REVIEW

**Comment by mmarod** [comment]
This actually only seems to work when there are more than 2 modules to position. I tried with exactly 2 modules and it still is not ordering properly...

File: spec/unit/puppet/provider/pam/augeas_spec.rb [hunk 1]
```diff
@@ -131,6 +131,31 @@
                           :arguments => ["try_first_pass","retry=3","type="],})
     end
 
+    describe "when reodering settings" do
+      it "Changing the order of an entry" do
```
Comment (raphink) [inline spec/unit/puppet/provider/pam/augeas_spec.rb]: I would rather this says "should change the order of an entry"

File: spec/unit/puppet/provider/pam/augeas_spec.rb [hunk 2]
```diff
@@ -131,6 +131,31 @@
...
+        aug_open(target, "Pam.lns") do |aug|
+          expect(aug.get("./2/module")).to eq("pam_unix.so")
```
Comment (raphink) [inline spec/unit/puppet/provider/pam/augeas_spec.rb]: I don't think it's necessary to test the initial state, we should know it already.

File: lib/puppet/provider/pam/augeas.rb [hunk 3]
```diff
@@ -78,19 +107,21 @@ def self.instances
...
-    if PAM_POSITION_ALIASES[key].has_key?(identifier)
-      expr = PAM_POSITION_ALIASES[key][identifier]
-      expr = key ? expr % [type, value] : expr % [type]
-    else
-      # if the identifier is not in the mapping
-      # we assume that its an xpath and so
-      # join everything after the placement
-      identifier = position.split(/ /)[1..-1].join(" ")
-      expr = identifier
+    unless position.nil?
```
Comment (raphink) [inline lib/puppet/provider/pam/augeas.rb]: Could you reuse `self.position_path` here maybe?

File: lib/puppet/provider/pam/augeas.rb [hunk 4]
```diff
@@ -45,6 +45,35 @@ def self.target(resource = nil)
     end
   end
 
+  def self.position_path (position, type)
+    placement, identifier, value = position.split(/ /)
+    key = !!value
+    if PAM_POSITION_ALIASES[key].has_key? identifier
+      expr = PAM_POSITION_ALIASES[key][identifier]
+      expr = key ? expr % [type, value] : expr % [type]
+    else
```
Comment (raphink) [inline lib/puppet/provider/pam/augeas.rb]: I might have liked to keep the comment that was in the original piece of code here.

File: lib/puppet/provider/pam/augeas.rb [hunk 5]
```diff
@@ -45,6 +45,35 @@ def self.target(resource = nil)
...
+      expr = position.split(/ /)[1..-1].join(" ")
+    end
+    return expr, placement
+  end
+
+  def in_position?
+    unless resource[:position].nil?
+      aug = nil
```
Comment (raphink) [inline lib/puppet/provider/pam/augeas.rb]: What is the goal of initializing `aug` here?

File: lib/puppet/provider/pam/augeas.rb [hunk 6]
```diff
@@ -45,6 +45,35 @@ def self.target(resource = nil)
...
+      path, before = self.class.position_path(resource[:position], resource[:type])
+
+      augopen do |aug|
+        if before == 'before'
+          mpath = "#{resource_path}[following-sibling::#{path}]"
+        else
+          mpath = "#{resource_path}[preceding-sibling::#{path}]"
+        end
+
```
Comment (raphink) [inline lib/puppet/provider/pam/augeas.rb]: `augopen` could be called here to restrict the critical section.

**Comment by raphink** [comment]
Thanks for the changes. Can you just fix the details I pointed out and squash please?

**Comment by raphink** [comment]
Merged, thank you.

**Pull request merged**